lib/component-repository/src/routes/virtual-components/GetAction.js

```diff
@@ -1,3 +1,31 @@
+const mongoose = require('mongoose');
+const ComponentVersion = require('../../models/ComponentVersion');
+
 module.exports = async function (req, res, next) {
-    return next();
+  const { componentVersionId, actionName } = req.params;
+
+  if (!mongoose.Types.ObjectId.isValid(componentVersionId)) {
+    return res
+      .status(400)
+      .send({ errors: [{ message: 'Invalid id', code: 400 }] });
+  };
+
+  const action = await ComponentVersion.findOne(
+    {
+      _id: componentVersionId,
+      'actions.name': actionName,
+    },
+    {
+      'actions.$': 1,
+    }
+  ).lean();
+  if (!action) {
+    return res.status(404).send({
+      errors: [{ message: 'Action with this name could not be found!', code: 404 }],
+    });
+  };
+
+  req.logger.debug(action);
+  res.data = action;
```

Comment:
Could you just return the action object?
I mean:
you can rename action to componentVersion and then:
`componentVersion.actions[0];`

Reply:
Done